lib/borda_poll.rb

```diff
@@ -0,0 +1,46 @@
+require 'set'
+
+class BordaPoll
+  class InvalidCandidateError < StandardError
+  end
+  class MultipleVoteError < StandardError
+  end
+  class OverdueVoteError < StandardError
+  end
+
+  attr_reader :title, :candidates, :votes, :closing
+
+  def initialize(title, candidates, closing)
+    @title = title
+    @candidates = candidates
+    @closing = closing
+    @votes = []
+  end
+
+  def add_vote(vote)
+    unless (vote.candidates.to_set ^ @candidates.to_set).length == 0 then
+      raise InvalidCandidateError.new(vote.candidates)
+    end
+
+    if @votes.map { |element| element.voter }.include?(vote.voter) then
+      raise MultipleVoteError.new(vote)
+    end
+
+    if DateTime.now() > closing then
+      raise OverdueVoteError.new(closing)
+    end
+
+    @votes.push(vote)
+  end
+
+  def count_votes
+    votedCandidates = @votes.map { |element| element.candidates }
```

Comment:
ruby の変数名はスネークケースが慣習です（JS と同時に書いたりしてるとよくやる）。

```suggestion
    voted_candidates = @votes.map(&:candidates)
```

Reply:
これ知らなかったです。
こういうRubyのスタイルに関する指摘はありがたいです。